### ISSUE TITLE
Setting principal key should run fk discovery convention

### ISSUE THREAD

**Issue by smitpatel** [aspnet/EntityFramework#4451]
Setting principal key determines principal/dependent ends explicitly which can help in figure out ambiguity. This should trigger foreignkey property discovery convention so that appropriate type from dependent side can be used as fk property.

**Comment by rowanmiller** [comment]
@smitpatel should this be closed now?

**Comment by smitpatel** [comment]
Previously, our ConfigurationSource only told us if something can be changed or not. Changing CS did not change the model. So if you set something which already set the same in the model we only upgraded the CS. So setting principal key on a relationship which has same principal key upgraded CS if needed.
Since we stopped guessing principal/dependent ends, setting principal key determines the principal ends and requires us to make appropriate changes to the relationship since the principal end is set now.
This would happen if the principal key is different than the existing one but if the existing principal key is the same then we just upgrade source and do nothing else.
